### PR TITLE
Tidy-up reference handling.

### PR DESCRIPTION
**Pull request by jeromekelleher**
- Added general range checking for ListReferenceBases, and raise a HTTP
  416 when these are out of bounds.
- Updated the object models for ReferenceSet and References, and
  generally refactored reference.py.
- Updated the simulated object models, and tested the translation of
  these through the protocol.
- Added more tests for ListReferenceBases.

### CODE REVIEW

File: ga4gh/datamodel/__init__.py [hunk 1]
```diff
@@ -381,18 +381,6 @@ def sanitizeAlignmentFileFetch(
         return referenceName, start, end
 
     @classmethod
-    def sanitizeFastaFileFetch(cls, start=None, end=None):
```
Comment (dcolligan) [inline ga4gh/datamodel/__init__.py]: why are we removing this?  it looks like we're still calling into fastaFile.fetch ...

**Comment by dcolligan** [comment]
looks good, generally

File: ga4gh/datamodel/__init__.py (hunk 1 above)
Comment (jeromekelleher) [inline ga4gh/datamodel/__init__.py]: There didn't seem to be much point to it --- ``checkQueryRange`` makes sure the values make sense and are in the correct range, so we'd just be doing the same thing again with the sanitize call. The values are already guaranteed to be ints, and they're in the correct range, so we're as safe as we're going to get.

**Comment by jeromekelleher** [comment]
@dcolligan --- I've pushed an update on this to include some test cases for ListReferenceBases paging.

**Comment by dcolligan** [comment]
+1

**Comment by jeromekelleher** [comment]
@dcolligan, merge me!

**Pull request merged**